src/zscale.jl

```diff
@@ -0,0 +1,248 @@
+    minpix = maximum([MIN_NPIXELS, Int128(npix * MAX_REJECT)])
+    last_ngoodpix = npix + 1
+
+    # This is the mask used in k-sigma clipping. 0 is good, 1 is bad
+    badpix = zeros(Int128, npix)
+
+    # Iterate
+    for niter in 1:maxiter
+        if (ngoodpix >= last_ngoodpix) || (ngoodpix < minpix)
+            break
+        end
+
+        # Accumulate sums to calculate straight line fit
+        sumx = 0
+        sumxx = 0
+        sumy = 0
+        sumxy = 0
+        sum = 0
+        for i in 1:length(badpix)
+            if badpix[i] == GOOD_PIXEL
+                sumx += xnorm[i]
+                sumxx += xnorm[i] * xnorm[i]
+                sumy += samples[i]
+                sumxy += sample[i] * xnorm[i]
+                sum += 1
+            end
+        end
+
+        delta = (sum * sumxx) - (sumx * sumx)
+        # Slope and intercept
+        intercept = ((sumxx * sumy) - (sumx * sumxy)) / delta
+        slope = ((sum * sumxy) - (sumx * sumy)) / delta
+
+        # Subtract fitted line from the data array
+        fitted = xnorm * slope + intercept
+        flat = samples - fitted
+
+        # Compute the k-sigma rejection threshold
+        ngoodpix, mean, sigma = zsc_compute_sigma(flat, badpix, npix)
+
+        threshold = sigma * krej
+
+        # Detect and reject pixels further than k*sigma from the fitted line
+        lcut = -threshold
+        hcut = threshold
+
+        for i in 1:npix
+            if (flat[i] < lcut) || (flat[i] > hcut)
+                badpix[i] = BAD_PIXEL
+            end
+        end
+        
+        # Convolve with a kernel of length ngrow
+        kernel = zeros(Int128, ngrow) .+ 1
+        badpix = convolve_same(badpix, kernel)
+
+        ngoodpix = 0
+        for i in 1:length(badpix)
+            if badpix[i] == GOOD_PIXEL
+                ngoodpix += 1
+            end
+        end
+    end
+
+    # Transform the line coefficients back to the X range [0:npix-1]
+    zstart = intercept - slope
+    zslope = slope * xscale
+
+    return ngoodpix, zstart, zslope
+end
+
+
+function zscale(image, nsamples=1000, contrast=0.25)
+
+    #=
+    Implement IRAF zscale algorithm
+    Parameters
+    ----------
+    image : arr
+        2-d numpy array
+    nsamples : int (Default: 1000)
+        Number of points in array to sample for determining scaling factors
+    contrast : float (Default: 0.25)
+        Scaling factor for determining min and max. Larger values increase the
+        difference between min and max values used for display.
+
+    Returns
+    -------
+    (z1, z2)
+    =#
+
+    # Sample the image
+    nsamples = minimum([nsamples, length(image)])
+    samples = zsc_sample(image, nsamples)
+    npix = length(samples)
+    sort!(samples)
+    
+    zmin = samples[1]
+    zmax = samples[end]
+
+    # For a one-indexed array
+    if npix % 2 == 1
+        center_pixel = Int128(round((npix + 1) / 2))
+        median = samples[center_pixel]
+    else
+        Int128(round(npix / 2))
```

Comment:
What is this line doing?

Reply:
It is supposed to be:
`center_pixel = Int128(round(npix / 2))`
Will rectify it with other changes.